src/lib/src/signature/simple.rs

```diff
@@ -74,6 +74,7 @@ impl PublicKey {
         reader: &mut impl Read,
         detached_signature: Option<&[u8]>,
     ) -> Result<(), WSError> {
+        let _header = Module::stream_init(reader)?;
```

Comment:
I guess this is the line that was missing?

Reply:
So it seems to be missing in the other verify() function in the same file.

Reply:
And in verify_multi?

Reply:
And in verify_matrix()? In fact I don't understand why this was required just in that specific location?

Reply:
You're right! It is missing in the other functions. 
This line is required because it reads the first few bytes from the reader (thus removing them from the buffer) which in turn leaves just the sections. Only then, can these sections be extracted into a SectionIterator object.